pkg/cli/cli_e2e_test.go

```diff
@@ -0,0 +1,165 @@
+
+	// read reports
+	truthReport := readReport(t, reportPath)
+	generatedReport := readReport(t, filepath.Join(generatedReportDir, reportFileName))
+	removeRelativePath(generatedReport, inputDir)
+	// compare reports
+	verifyReport(t, inputDir, truthReport, generatedReport)
+	verifyObfuscation(t, outputDir, generatedReport)
+}
+
+func removeRelativePath(r *reporting.Report, path string) {
+	for i := range r.Omissions {
+		r.Omissions[i] = strings.TrimPrefix(r.Omissions[i], path)
+	}
+}
+
+func verifyObfuscation(t *testing.T, dir string, report *reporting.Report) {
+	// report should already be verified by verifyReport before to ensure it does contain correct information
+	generatedMap := map[string]string{}
+	for _, obfuscator := range report.Replacements {
+		for _, replacement := range obfuscator {
+			for _, occurrence := range replacement.Occurrences {
+				generatedMap[occurrence.Original] = replacement.ReplacedWith
+			}
+		}
+	}
+	err := filepath.Walk(dir, func(path string, info fs.FileInfo, err error) error {
+		if err != nil {
+			return nil
+		}
+
+		for k, v := range generatedMap {
+			assert.NotContainsf(t, path, k, "path should not contain secret IP %s, but rather its replacement %s", k, v)
+		}
+
+		if !info.IsDir() {
+			file, err := ioutil.ReadFile(path)
+			if err != nil {
+				return err
+			}
+
+			content := string(file)
+			for k, v := range generatedMap {
+				assert.NotContainsf(t, content, k, "file '%s' should not contain secret IP %s, but rather its replacement %s", path, k, v)
+			}
+		}
+
+		return nil
+	})
+
+	require.NoError(t, err)
+}
+
+func verifyReport(t *testing.T, inputDir string, truthReport *reporting.Report, generatedReport *reporting.Report) {
+	verifyReplacements(t, inputDir, truthReport, generatedReport)
+	verifyOmissions(t, inputDir, truthReport, generatedReport)
+}
+
+func verifyReplacements(t *testing.T, inputDir string, truthReport *reporting.Report, generatedReport *reporting.Report) {
+	tr := reportInternalRepresentation(truthReport)
+	gr := reportInternalRepresentation(generatedReport)
+	replacementReportsMatch(t, tr, gr)
+}
+
+func verifyOmissions(t *testing.T, inputDir string, truthReport *reporting.Report, generatedReport *reporting.Report) {
+	sort.Strings(truthReport.Omissions)
+	sort.Strings(generatedReport.Omissions)
+	assert.Equal(t, truthReport.Omissions, generatedReport.Omissions)
+}
+
+func reportInternalRepresentation(report *reporting.Report) obfuscator.ReplacementReport {
+	var repls []obfuscator.Replacement
+	for _, o := range report.Replacements {
+		for _, r := range o {
+			count := map[string]uint{}
+			for _, c := range r.Occurrences {
+				count[c.Original] = c.Count
+			}
+			repls = append(repls, obfuscator.Replacement{
+				Canonical:    r.Canonical,
+				ReplacedWith: r.ReplacedWith,
+				Counter:      count,
+			})
+		}
+	}
+	return obfuscator.ReplacementReport{Replacements: repls}
+}
+
+func replacementReportsMatch(t *testing.T, want, got obfuscator.ReplacementReport) {
```

Comment:
can we reuse that report matching from your testutils? moved it into the reporting package I believe

Reply:
slightly different - we don't want to check `replacedWith` as that changes between reports (which is expected)